### PR TITLE
[SPARK-48028][TESTS] Regenerate benchmark results after turning ANSI on

### PR DESCRIPTION
**Pull request by yaooqinn**
### What changes were proposed in this pull request?

This PR aims to fix benchmark errors and regenerate benchmark results for Apache Spark 4.0.0 after turning ANSI on.

The latest baseline has been updated by SPARK-47513.

Benchmarks fixed:
- AvroReadBenchmark
- DataSourceReadBenchmark
- DateTimeBenchmark (Fixed before this PR)
- InExpressionBenchmark
- MetadataStructBenchmark
- OrcReadBenchmark
- TPCDSQueryBenchmark <b><ANSI OFF</b>

### Why are the changes needed?

SPARK-44444 turns ANSI on by default, there could be performance related issues

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Manual review. / [Run benchmarks](https://github.com/yaooqinn/spark/actions/workflows/benchmark.yml)

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

**Comment by yaooqinn** [comment]
Waiting for CI results

https://github.com/yaooqinn/spark/actions/runs/8867970355
https://github.com/yaooqinn/spark/actions/runs/8867971118

**Comment by dongjoon-hyun** [comment]
Thank you!

**Comment by dongjoon-hyun** [comment]
Looking forward to seeing the result.

**Comment by yaooqinn** [comment]
Thank you @dongjoon-hyun.

Unfortunately, they timed out.

**Comment by yaooqinn** [comment]
Pending CI results


https://github.com/yaooqinn/spark/actions/runs/8872679083
https://github.com/yaooqinn/spark/actions/runs/8872179637

**Comment by yaooqinn** [comment]
Witnessed an extremely slow benchmark https://github.com/apache/spark/pull/45453#issuecomment-2082071977

**Comment by yaooqinn** [comment]
There is a failure in TPCDS query 90:

```
Running benchmark: TPCDS
  Running case: q90
24/04/29 22:09:37 ERROR Utils: Aborting task
org.apache.spark.SparkArithmeticException: [DIVIDE_BY_ZERO] Division by zero. Use `try_divide` to tolerate divisor being 0 and return NULL instead. If necessary set "spark.sql.ansi.enabled" to "false" to bypass this error. SQLSTATE: 22012
== SQL (line 1, position 8) ==
SELECT cast(amc AS DECIMAL(15, 4)) / cast(pmc AS DECIMAL(15, 4)) am_pm_ratio
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

	at org.apache.spark.sql.errors.QueryExecutionErrors$.divideByZeroError(QueryExecutionErrors.scala:203)
	at
```

In this PR, I tend to disable ansi for tpcds benchmark temporarily.

File: sql/core/benchmarks/AggregateBenchmark-results.txt [hunk 1]
```diff
@@ -2,147 +2,147 @@
...
 
-OpenJDK 64-Bit Server VM 17.0.10+7-LTS on Linux 6.5.0-1016-azure
+OpenJDK 64-Bit Server VM 17.0.11+9-LTS on Linux 6.5.0-1018-azure
 AMD EPYC 7763 64-Core Processor
 agg w/o group:                            Best Time(ms)   Avg Time(ms)   Stdev(ms)    Rate(M/s)   Per Row(ns)   Relative
 ------------------------------------------------------------------------------------------------------------------------
-agg w/o group wholestage off                      30915          32941        2865         67.8          14.7       1.0X
-agg w/o group wholestage on                         717            720           2       2924.3           0.3      43.1X
+agg w/o group wholestage off                      38161          38820         933         55.0          18.2       1.0X
+agg w/o group wholestage on                        2472           2488          10        848.5           1.2      15.4X
```
Comment (yaooqinn) [inline sql/core/benchmarks/AggregateBenchmark-results.txt:10]: mark

File: sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/TPCDSQueryBenchmark.scala [hunk 2]
```diff
@@ -58,6 +58,7 @@ object TPCDSQueryBenchmark extends SqlBasedBenchmark with Logging {
       .set("spark.sql.crossJoin.enabled", "true")
       .set("spark.serializer", "org.apache.spark.serializer.KryoSerializer")
       .set("spark.kryo.registrationRequired", "true")
+      .set("spark.sql.ansi.enabled", "false")
```
Comment (yaooqinn) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/TPCDSQueryBenchmark.scala:61]: cc @dongjoon-hyun @cloud-fan Now we seem to lose TPCDS compliance by default.

Comment (dongjoon-hyun) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/TPCDSQueryBenchmark.scala:61]: Just for my understanding, TPCDS data have some overflows?

File: connector/avro/src/test/scala/org/apache/spark/sql/execution/benchmark/AvroReadBenchmark.scala [hunk 3]
```diff
@@ -87,7 +87,7 @@ object AvroReadBenchmark extends SqlBasedBenchmark {
 
         prepareTable(
           dir,
-          spark.sql("SELECT CAST(value AS INT) AS c1, CAST(value as STRING) AS c2 FROM t1"))
+          spark.sql(s"SELECT value % ${Int.MaxValue} AS c1, CAST(value as STRING) AS c2 FROM t1"))
```
Comment (dongjoon-hyun) [inline connector/avro/src/test/scala/org/apache/spark/sql/execution/benchmark/AvroReadBenchmark.scala:90]: This looks like a different benchmark, why do we use a static value here, @yaooqinn ?

File: connector/avro/src/test/scala/org/apache/spark/sql/execution/benchmark/AvroReadBenchmark.scala [hunk 4]
```diff
@@ -106,7 +106,8 @@ object AvroReadBenchmark extends SqlBasedBenchmark {
         import spark.implicits._
         spark.range(values).map(_ => Random.nextLong()).createOrReplaceTempView("t1")
 
-        prepareTable(dir, spark.sql("SELECT value % 2 AS p, value AS id FROM t1"), Some("p"))
+        prepareTable(dir,
+          spark.sql(s"SELECT value % 2 AS p, value % ${Int.MaxValue} AS id FROM t1"), Some("p"))
```
Comment (dongjoon-hyun) [inline connector/avro/src/test/scala/org/apache/spark/sql/execution/benchmark/AvroReadBenchmark.scala:110]: ditto.

File: connector/avro/src/test/scala/org/apache/spark/sql/execution/benchmark/AvroReadBenchmark.scala [hunk 5]
```diff
@@ -176,7 +177,7 @@ object AvroReadBenchmark extends SqlBasedBenchmark {
       withTempTable("t1", "avroTable") {
         import spark.implicits._
         val middle = width / 2
-        val selectExpr = (1 to width).map(i => s"value as c$i")
+        val selectExpr = (1 to width).map(i => s"value % ${Int.MaxValue} as c$i")
```
Comment (dongjoon-hyun) [inline connector/avro/src/test/scala/org/apache/spark/sql/execution/benchmark/AvroReadBenchmark.scala:180]: ditto.

File: sql/core/benchmarks/DataSourceReadBenchmark-jdk21-results.txt [hunk 6]
```diff
@@ -2,430 +2,430 @@
...
-SQL Parquet Vectorized: DataPageV2                  268            300          20         58.7          17.0      54.5X
-SQL Parquet MR: DataPageV1                         2289           2314          36          6.9         145.5       6.4X
-SQL Parquet MR: DataPageV2                         1993           1995           3          7.9         126.7       7.3X
-SQL ORC Vectorized                                  215            224          12         73.1          13.7      68.0X
-SQL ORC MR                                         1840           1851          17          8.6         117.0       7.9X
-
-OpenJDK 64-Bit Server VM 21.0.2+13-LTS on Linux 6.5.0-1016-azure
+SQL CSV                                           11095          11134          54          1.4         705.4       1.0X
+SQL Json                                           9688           9701          18          1.6         616.0       1.1X
+SQL Parquet Vectorized: DataPageV1                  293            297           4         53.7          18.6      37.9X
```
Comment (dongjoon-hyun) [inline sql/core/benchmarks/DataSourceReadBenchmark-jdk21-results.txt:99]: `DataPageV1` seems to show a regression in `BIGINT` column. It's slower than `DataPageV2` of Parquet.

File: sql/core/benchmarks/DataSourceReadBenchmark-jdk21-results.txt [hunk 7]
```diff
@@ -2,430 +2,430 @@
...
-SQL Parquet Vectorized: DataPageV2                  161            167           5         97.4          10.3      90.5X
-SQL Parquet MR: DataPageV1                         2239           2249          14          7.0         142.4       6.5X
-SQL Parquet MR: DataPageV2                         2125           2169          63          7.4         135.1       6.9X
-SQL ORC Vectorized                                  352            366          11         44.6          22.4      41.5X
-SQL ORC MR                                         1823           1824           1          8.6         115.9       8.0X
-
-OpenJDK 64-Bit Server VM 21.0.2+13-LTS on Linux 6.5.0-1016-azure
+SQL CSV                                           11485          11545          86          1.4         730.2       1.0X
+SQL Json                                          11591          11597           8          1.4         737.0       1.0X
+SQL Parquet Vectorized: DataPageV1                  269            288          18         58.5          17.1      42.7X
```
Comment (dongjoon-hyun) [inline sql/core/benchmarks/DataSourceReadBenchmark-jdk21-results.txt:143]: This also has slightly different ratio.

File: sql/core/benchmarks/DataSourceReadBenchmark-jdk21-results.txt [hunk 8]
```diff
@@ -2,430 +2,430 @@
...
-SQL Parquet Vectorized: DataPageV2 (Nested Column Disabled)           2816           2821           7          5.6         179.0       0.8X
-SQL Parquet Vectorized: DataPageV2 (Nested Column Enabled)             332            339           7         47.3          21.1       6.8X
-
-OpenJDK 64-Bit Server VM 21.0.2+13-LTS on Linux 6.5.0-1016-azure
+SQL ORC MR                                                            2207           2213           9          7.1         140.3       1.0X
+SQL ORC Vectorized (Nested Column Disabled)                           2245           2267          32          7.0         142.7       1.0X
+SQL ORC Vectorized (Nested Column Enabled)                             285            305          38         55.1          18.1       7.7X
+SQL Parquet MR: DataPageV1                                            2807           2812           8          5.6         178.4       0.8X
+SQL Parquet Vectorized: DataPageV1 (Nested Column Disabled)           3405           3407           3          4.6         216.5       0.6X
+SQL Parquet Vectorized: DataPageV1 (Nested Column Enabled)             291            321          20         54.0          18.5       7.6X
```
Comment (dongjoon-hyun) [inline sql/core/benchmarks/DataSourceReadBenchmark-jdk21-results.txt:215]: This also shows a regression on `DataPageV1` (according to the ratio change of `DataPageV1` vs `DataPageV2`. `DataPageV1` becomes slower than `DataPageV2`.

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/OrcReadBenchmark.scala [hunk 9]
```diff
@@ -79,6 +79,12 @@ object OrcReadBenchmark extends SqlBasedBenchmark {
     spark.read.format(HIVE_ORC_FORMAT).load(dirORC).createOrReplaceTempView("hiveOrcTable")
   }
 
+  private def getExpr(dataType: DataType = IntegerType): String = dataType match {
+    case ByteType => "cast(value % 128 as byte)"
+    case ShortType => "cast(value % 32768 as short)"
+    case _ => s"cast(value % ${Int.MaxValue} as ${dataType.sql})"
+  }
```
Comment (dongjoon-hyun) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/OrcReadBenchmark.scala:86]: Thank you!

**Review by dongjoon-hyun — COMMENTED** [review]
Thank you, @yaooqinn . This is a nice data point.

File: connector/avro/src/test/scala/org/apache/spark/sql/execution/benchmark/AvroReadBenchmark.scala (hunk 3 above)
Comment (dongjoon-hyun) [reply connector/avro/src/test/scala/org/apache/spark/sql/execution/benchmark/AvroReadBenchmark.scala:90]: Oh, nvm.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Comment by dongjoon-hyun** [comment]
Merged to master.

File: sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/TPCDSQueryBenchmark.scala (hunk 2 above)
Comment (yaooqinn) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/TPCDSQueryBenchmark.scala:61]: DIVIDE_BY_ZERO in q90 so far

Comment (yaooqinn) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/TPCDSQueryBenchmark.scala:61]: https://issues.apache.org/jira/browse/SPARK-48066 has been filed

**Comment by yaooqinn** [comment]
Thank you @dongjoon-hyun

File: sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/TPCDSQueryBenchmark.scala (hunk 2 above)
Comment (dongjoon-hyun) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/TPCDSQueryBenchmark.scala:61]: Thank you!

**Pull request closed without merging**